File: Task/Least-common-multiple/PascalABC.NET/least-common-multiple.pas
function GCD(a,b: integer): integer;
begin
  while b > 0 do
    (a,b) := (b,a mod b);
  Result := a;
end;

function LCM(a,b: integer): integer := a = 0 ? 0 : a div GCD(a,b) * b;

begin
  Println(LCM(12,18));
end.

File: Task/Least-common-multiple/Euphoria/least-common-multiple.ex
function gcd(integer m, integer n)
    integer tmp
    while m do
        tmp = m
        m = remainder(n,m)
        n = tmp
    end while
    return n
end function

function lcm(integer m, integer n)
    return m / gcd(m, n) * n
end function
